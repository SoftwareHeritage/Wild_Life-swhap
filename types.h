/* Copyright 1991 Digital Equipment Corporation.
** All Rights Reserved.
*****************************************************************/

#ifndef _TYPES_H_
#define _TYPES_H_

extern int types_modified;
extern int type_count;

extern int redefine();

extern void assert_type();
extern void assert_complicated_type();
extern void print_codes();

extern void assert_protected();
extern void assert_delay_check();
extern void assert_args_not_eval();
extern void clear_already_loaded();
extern void inherit_always_check();

extern int glb();
extern int glb_code();
extern int glb_value();
extern int sub_type();
extern int overlap_type();
extern int matches();

extern ptr_int_list decode();

extern void print_def_type();

ptr_int_list lub();
void or_codes();
ptr_int_list copyTypeCode();

#endif
