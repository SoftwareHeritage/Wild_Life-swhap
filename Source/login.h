/* Copyright 1991 Digital Equipment Corporation.
** All Rights Reserved.
*****************************************************************/

/* High level calls */
extern void assert_clause();
extern void prove_psi_term();

extern void merge_unify();

extern void get_one_arg();
extern void get_one_arg_addr();
extern void get_two_args();

extern void fetch_def();

/* Trailing */
extern ptr_stack undo_stack;
extern void push_ptr_value();
extern void push_ptr_value_global();
extern void push2_ptr_value();
extern void push_window();
extern void clean_undo_window();

#ifdef TS
extern void push_psi_ptr_value(); /* 9.6 */
extern unsigned long global_time_stamp; /* 9.6 */
/* Trail if q was last modified before the topmost choice point */
#define TRAIL_CONDITION(Q) (choice_stack && \
                            choice_stack->time_stamp>=Q->time_stamp)
#endif

/* Detrailing */
extern void undo();
extern void undo_actions();

/* User-interface */
extern long stepflag;
extern long ignore_eff;
extern long goal_count;
extern void show_count();
extern struct tms start_time,end_time;
