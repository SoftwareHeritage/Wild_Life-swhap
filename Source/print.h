/* Copyright 1991 Digital Equipment Corporation.
** All Rights Reserved.
*****************************************************************/


typedef struct _tab_brk *       ptr_tab_brk;
typedef struct _item *          ptr_item;

typedef struct _tab_brk {
  long column;
  long broken;
  long printed;
} tab_brk;

typedef struct _item {
  char *str;
  ptr_tab_brk tab;
} item;

extern void init_print();
extern void pred_write();
extern void listing_pred_write();
extern long str_to_int();

extern long print_variables();
extern void print_resid_message();
extern void print_operator_kind();

extern void display_psi();
extern void display_psi_stdout();
extern void display_psi_stream();
extern void display_psi_stderr();

extern void print_code();

extern char *no_name;
extern char *buffer;

/* Global flags that modify how writing is done. */
extern long print_depth;
extern long indent;
extern long const_quote;
extern long write_stderr;
extern long write_corefs;
extern long write_resids;
extern long write_canon;
