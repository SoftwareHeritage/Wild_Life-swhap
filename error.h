/* Copyright 1991 Digital Equipment Corporation.
** All Rights Reserved.
**
*****************************************************************/

#ifndef _ERROR_H_
#define _ERROR_H_

extern void stack_info();

extern void init_trace();
extern void reset_step();
extern void tracing();
extern void new_trace();
extern void new_step();
extern void toggle_trace();
extern void toggle_step();
extern int quietflag; /* 21.1 */
extern int trace;
extern int verbose; /* 21.1 */
extern int stepflag;
extern int steptrace;
extern int stepcount;

#define NOTQUIET (!quietflag || verbose) /* 21.1 */

extern int warning();
extern int warningx();
extern void perr();
extern void perr_s();
extern void perr_s2();
extern void perr_i();

extern void report_error();
extern void report_warning();
extern void report_error2();
extern void report_warning2();

extern void nonnum_warning();
extern int bit_and_warning();
extern int bit_or_warning();
extern int mod_warning();
extern int shift_warning();

#ifndef NOTRACE
#define Traceline  if (trace) traceline
#else
#define Traceline  if (0) traceline
#endif

/* 21.1 */
#define Infoline   if (NOTQUIET) infoline

extern void Errorline();
extern void Syntaxerrorline();
extern void outputline(); /* To output_stream */
#define Warningline if (warningflag) warningline

#endif
