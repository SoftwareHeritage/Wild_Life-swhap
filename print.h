/* Copyright 1991 Digital Equipment Corporation.
** All Rights Reserved.
*****************************************************************/


typedef struct _tab_brk *       ptr_tab_brk;
typedef struct _item *          ptr_item;

typedef struct _tab_brk {
  int column;
  int broken;
  int printed;
} tab_brk;

typedef struct _item {
  char *str;
  ptr_tab_brk tab;
} item;

extern void init_print();
extern void pred_write();
extern void listing_pred_write();
extern int str_to_int();

extern int print_variables();
extern void print_resid_message();
extern void print_operator_kind();

extern void display_psi();
extern void display_psi_stdout();
extern void display_psi_stream();
extern void display_psi_stderr();

extern void print_code();

extern char *no_name;
extern char *buffer;

/* Global flags that modify how writing is done. */
extern int print_depth;
extern int indent;
extern int const_quote;
extern int write_stderr;
extern int write_corefs;
extern int write_resids;
