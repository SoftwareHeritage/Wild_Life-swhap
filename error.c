/* Copyright 1991 Digital Equipment Corporation.
 * All Rights Reserved.
 *
 * History:
 *  SCG  21  Tue Jun  2 14:15:36 1992
 *    added newTrace which allows a trace line to be one function call
 *  SCG  14  Wed May 27 13:37:51 1992
 *    added reportAndAbort() which is like report_error followed by
 *    an c_abort.
*****************************************************************/

#include "extern.h"
#include "print.h"
#include "types.h"
#include "login.h"
#include "error.h"

int warningflag=TRUE;
int trace=FALSE;
int stepflag;
int steptrace;
int stepcount;


/* Depth of goal stack */
static int depth_gs()
{
  int i=0;
  ptr_goal g=goal_stack;

  while (g) { i++; g=g->next; }
  return i;
}


/* Depth of choice point stack */
static int depth_cs()
{
  int i=0;
  ptr_choice_point c=choice_stack;

  while (c) { i++; c=c->next; }
  return i;
}


/* Depth of trail (undo) stack */
static int depth_ts()
{
  ptr_stack t=undo_stack;
  int i=0;

  while (t) { i++; t=t->next; }
  return i;
}


void stack_info(outfile)
FILE *outfile;
{
  /* Information about size of embedded stacks */
  if (verbose) {
    int gn,cn,tn;
    fprintf(outfile,"*** Stack depths [");
    fprintf(outfile,"%d goal%s, %d choice point%s, %d trail entr%s",
            gn=depth_gs(),(gn!=1?"s":""),
            cn=depth_cs(),(cn!=1?"s":""),
            tn=depth_ts(),(tn!=1?"ies":"y"));
    fprintf(outfile,"]\n");
  }
}



void vinfoline ARGS((char *format, VarArgBaseDecl));

void outputline(format, VarArgBase)
char *format;
VarArgBaseDecl;
{
  VarArgDecl;

  VarArgInit(format);
  vinfoline(format, output_stream, VarArg);
}

void traceline(format, VarArgBase)
char *format;
VarArgBaseDecl;
{
  VarArgDecl;

  if ((trace == 2) && (format[0] != 'p')) return;
  tracing();
  VarArgInit(format);
  vinfoline(format, stdout, VarArg);
}

void infoline(format, VarArgBase)
char *format;
VarArgBaseDecl;
{
  VarArgDecl;

  VarArgInit(format);
  vinfoline(format, stdout, VarArg);
}

void warningline(format, VarArgBase)
char *format;
VarArgBaseDecl;
{
  VarArgDecl;

  VarArgInit(format);
  fprintf(stderr,"*** Warning: ");
  vinfoline(format, stderr, VarArg);
}

/* New error printing routine */
void Errorline(format, VarArgBase)
char *format;
VarArgBaseDecl;
{
  VarArgDecl;

  VarArgInit(format);
  fprintf(stderr,"*** Error: ");
  vinfoline(format, stderr, VarArg);
}

void Syntaxerrorline(format, VarArgBase)
char *format;
VarArgBaseDecl;
{
  VarArgDecl;

  VarArgInit(format);
  fprintf(stderr,"*** Syntax error: ");
  vinfoline(format, stderr, VarArg);
}

void vinfoline(format, outfile, VarArg)
char *format;
FILE *outfile;
VarArgDecl;
{
  char *p;
  ptr_psi_term psi;
  char buffer[5];
  ptr_int_list pil;
  operator kind;
  def_type t;
  
  for (p=format; *p; p++)
  {
    if (*p == '%')
    {
      switch (*++p)
      {
      case 'd':
      case 'x':
      case 's':
        buffer[0] = '%';
        buffer[1] = *p;
        buffer[2] = 0;
        vfprintf(outfile, buffer, VarArg);
        VarArgNext(int);
        break;
            
      case 'C':
        /* type coding as bin string */
        pil = VarArgNext(ptr_int_list);
        print_code(outfile,pil);
        break;
        
      case 'P':
        psi = VarArgNext(ptr_psi_term);
        display_psi(outfile,psi);
        break;

      case 'O':
        kind = VarArgNext(operator);
        print_operator_kind(outfile,kind);
        break;

      case 'T':
        assert(outfile==stderr);
        t = VarArgNext(def_type);
        print_def_type(t);
        break;

      case 'E':
        assert(outfile==stderr);
        psi_term_error();
        break;
        
      case '%':
        putc(*p,outfile);
        break;
        
      default:
        fprintf(outfile,"<%c follows %% : report bug >", *p);
        break;
      }
    }
    else
      putc(*p,outfile);
  }
  VarArgEnd();
}

/********************************************************************/

/* Utilities for tracing and single stepping */

/* Initialize all tracing variables */
void init_trace()
{
  trace=FALSE;
  stepflag=FALSE;
  stepcount=0;
}

/* Reset stepcount to zero */
/* Should be called when prompt is printed */
void reset_step()
{
  if (stepcount>0) {
    stepcount=0;
    stepflag=TRUE;
  }
}

void tracing()
{
  int i;
  int indent;

  printf("T%04ld",goal_count);
  printf(" C%02ld",depth_cs());
  indent=depth_gs();
  if (indent>=MAX_TRACE_INDENT) printf(" G%02ld",indent);
  indent = indent % MAX_TRACE_INDENT;
  for (i=indent; i>=0; i--) printf(" ");
  steptrace=TRUE;
}


void new_trace(newtrace)
int newtrace;
{
  trace = newtrace;
  printf("*** Tracing is turned ");
  printf(trace?"on.":"off.");
  if (trace == 2) printf(" Only for Proves");
  printf("\n");
}

void new_step(newstep)
int newstep;
{
  stepflag = newstep;
  printf("*** Single stepping is turned ");
  printf(stepflag?"on.\n":"off.\n");
  new_trace(stepflag);
  steptrace=FALSE;
}

void set_trace_to_prove()
{
  new_trace(2);
}

void toggle_trace()
{
  new_trace(trace?0:1);
}


void toggle_step()
{
  new_step(!stepflag);
}

/********************************************************************/

/* Old error printing routines -- these should be superceded by Errorline */

void perr(str)
char *str;
{
  fprintf(stderr,str);
}

void perr_s(s1,s2)
char *s1,*s2;
{
  fprintf(stderr,s1,s2);
}

void perr_s2(s1,s2,s3)
char *s1,*s2,*s3;
{
  fprintf(stderr,s1,s2,s3);
}

void perr_i(str,i)
char *str;
int i;
{
  fprintf(stderr,str,i);
}


int warning()
{
  if (warningflag) perr("*** Warning: ");
  return warningflag;
}


int warningx()
{
  if (warningflag) perr("*** Warning");
  return warningflag;
}


/* Main routine for report_error and report_warning */
void report_error_main(g,s,s2)
ptr_psi_term g;
char *s, *s2;
{
  FILE *f;

  perr_s2("*** %s: %s in '",s2,s);
  display_psi_stderr(g);
  perr("'.\n");
}



/******** REPORT_ERROR(g,s)
  Print an appropriate error message. G is the
  psi-term which caused the error, S a message to print.
  Format: '*** Error: %s in 'g'.'
*/
void report_error(g,s)
ptr_psi_term g;
char *s;
{
  report_error_main(g,s,"Error");
}



/******** REPORTANDABORT(g,s)
  Print an appropriate error message. G is the
  psi-term which caused the error, S a message to print.
  Format: '*** Error: %s in 'g'.'
*/
int reportAndAbort(g,s)
ptr_psi_term g;
char *s;
{
  report_error_main(g,s,"Error");
  return abort_life();
}


/******** REPORT_WARNING(g,s)
  Print an appropriate error message. G is the
  psi-term which caused the error, S a message to print.
  Format: '*** Warning: %s in 'g'.'
*/
void report_warning(g,s)
ptr_psi_term g;
char *s;
{
  if (warningflag) report_error_main(g,s,"Warning");
}


/* Main routine for report_error2 and report_warning2 */
void report_error2_main(g,s,s2)
ptr_psi_term g;
char *s, *s2;
{
  FILE *f;

  perr_s("*** %s: argument '",s2);
  display_psi_stderr(g);
  perr_s("' %s.\n",s);
}



/********* REPORT_ERROR2(g,s)
  Like report_error, with a slightly different format.
  Format: '*** Error: argument 'g' %s.'
*/
void report_error2(g,s)
ptr_psi_term g;
char *s;
{
  report_error2_main(g,s,"Error");
}



/********* REPORT_WARNING2(g,s)
  Like report_warning, with a slightly different format.
  Format: '*** Warning: argument 'g' %s.'
*/
void report_warning2(g,s)
ptr_psi_term g;
char *s;
{
  if (warningflag) report_error2_main(g,s,"Warning");
}



/* Give error message if there is an argument which cannot unify with */
/* a real number. */
void nonnum_warning(t,arg1,arg2)
ptr_psi_term t,arg1,arg2;
{
  if ((arg1 && !overlap_type(arg1->type,real)) ||
      (arg2 && !overlap_type(arg2->type,real))) {
    report_warning(t,"non-numeric argument(s)");
  }
}

/********************************************************************/

/* Error checking routines for bit_and, bit_or, shift, and modulo */

int nonint_warning(arg, val, msg)
ptr_psi_term arg;
REAL val;
char *msg;
{
  int err=FALSE;

  if (val!=floor(val)) {
    report_warning2(arg, msg);
    err=TRUE;
  }
  return err;
}

int bit_and_warning(arg, val)
ptr_psi_term arg;
REAL val;
{
  return nonint_warning(arg,val,"of bitwise 'and' operation is not an integer");
}

int bit_or_warning(arg, val)
ptr_psi_term arg;
REAL val;
{
  return nonint_warning(arg,val,"of bitwise 'or' operation is not an integer");
}

int mod_warning(arg, val)
ptr_psi_term arg;
REAL val;
{
  return nonint_warning(arg,val,"of modulo operation is not an integer");
}

int shift_warning(dir, arg, val)
int dir;
ptr_psi_term arg;
REAL val;
{
  if (dir)
    return nonint_warning(arg,val,"of right shift operation is not an integer");
  else
    return nonint_warning(arg,val,"of left shift operation is not an integer");
}

/********************************************************************/
