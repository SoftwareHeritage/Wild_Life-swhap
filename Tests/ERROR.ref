addtab.diff
anyorder.diff
apply.diff
arith1.diff
arith2.diff
arith3.diff
*** Error: division by zero in B:0 / B.
*** Error: division by zero in real / 0.
*** Error: division by zero in 2 / 0.
arith4.diff
arith5.diff
arith6.diff
*** Error: division by zero in real / 0.
*** Error: division by zero in 0 / 0.
*** Error: division by zero in 1 / 0.
arith7.diff
*** Error: division by zero in real / 0.
*** Error: division by zero in real / 0.
arith8.diff
arith9.diff
*** Garbage Collections: 2
assert1.diff
*** Garbage Collections: 1
assert2.diff
*** Garbage Collections: 6
assert3.diff
assert4.diff
bagof1.diff
bagof2.diff
bagof3.diff
bagof4.diff
bagof5.diff
bigstr.diff
bk_assign.diff
boites1.diff
boites2.diff
bruno_disj.diff
cb.diff
chapla.diff
choice.diff
clause.diff
colon.diff
compose.diff
constr.diff
copy1.diff
copy2.diff
copy3.diff
curry1.diff
*** Garbage Collections: 8
curry2.diff
curry3.diff
curry4.diff
curry5.diff
cut1.diff
cut2.diff
*** Garbage Collections: 4
cut3.diff
cycles1.diff
cycles2.diff
decl.diff
delay.diff
dichotomy.diff
dict.diff
disequality1.diff
disequality2.diff
disequality3.diff
disequality4.diff
disequality5.diff
disequality6.diff
disequality7.diff
disequality8.diff
disequality9.diff
disj1.diff
disj2.diff
disj3.diff
disj4.diff
disj5.diff
disj6.diff
*** Garbage Collections: 2
dissolve.diff
*** Error: the user-defined predicate 'p' may not be changed.
*** Error: the user-defined predicate 'p' may not be changed.
*** Error: the user-defined predicate 'p' may not be changed.
*** Error: the user-defined predicate 'p' may not be changed.
dynstat.diff
*** Syntax error near line 4: bad expression.
eoftest.diff
*** Garbage Collections: 12
fact.diff
fail.diff
feature.diff
freeze1.diff
*** Error: the argument @ < @ of freeze must be a predicate.
*** Error: the argument @ = @ of freeze must be user-defined.
freeze2.diff
freeze3.diff
*** Error: the argument var(@~) of freeze must be user-defined.
*** Error: the argument @~ < @ of freeze must be a predicate.
*** Error: the argument write(@) of freeze must be user-defined.
freeze4.diff
function1.diff
function2.diff
fusil.diff
*** Garbage Collections: 9
gauss.diff
*** Garbage Collections: 74
gctest.diff
genint.diff
glb.diff
glblub.diff
grammars.diff
hamming.diff
hanoi.diff
inherit.diff
intabs.diff
*** Warning: extending definition of sort 'int'.
*** Warning: extending definition of sort 'int'.
inttrace.diff
isatest.diff
junk.diff
lefun1.diff
lefun2.diff
list_bda.diff
listing1.diff
*** Warning: extending definition of sort 'int'.
*** Warning: extending definition of sort 'zero'.
*** Warning: extending definition of sort 'int'.
listing2.diff
loadquery.diff
long.diff
*** Garbage Collections: 12
longstr.diff
loop.diff
lub.diff
machine.diff
*** Garbage Collections: 3
magic.diff
makestr.diff
mort.diff
*** Error: [@|@] should be a nil-terminated list in mresiduate.
*** Error: {@;@;@} should be a nil-terminated list in mresiduate.
mresid.diff
nesting.diff
nicetest.diff
nonstrict1.diff
nonstrict2.diff
nonstrict3.diff
nonstrict4.diff
nonstrict5.diff
numeric.diff
op1.diff
*** Error: numbers or strings may not be operators in '*op*'(500,xfy,"aaa").
*** Error: numbers or strings may not be operators in '*op*'(500,xfy,123).
*** Error: numbers or strings may not be operators in '*op*'(500,xfy,123.3).
*** Error: bad operator kind 'xff'.
*** Error: precedence must range from 1 to 1200 in '*op*'(5000,xfy,asd).
*** Error: precedence must range from 1 to 1200 in '*op*'(0,xfy,asd).
*** Error: precedence must be a positive integer in '*op*'(- 234,xfy,asd).
*** Error: precedence must be a positive integer in '*op*'(abc,xfy,asd).
*** Error: argument missing in '*op*'(500,xfy).
*** Error: argument missing in '*op*'(500,3 => asd).
*** Error: argument missing in '*op*'(2 => xfy,3 => asd).
op2.diff
order.diff
parents.diff
parse.diff
part.diff
person1.diff
person2.diff
*** Warning: extending definition of sort 'int'.
posint.diff
preparser.diff
*** Warning: extending definition of sort 'int'.
*** Garbage Collections: 15
prime.diff
print1.diff
print2.diff
*** Error: argument in print_depth(...) must be positive or zero.
*** Error: argument in print_depth(-1) must be positive or zero.
print_depth1.diff
print_depth2.diff
printvar.diff
prove.diff
pyth.diff
*** Error: the user-defined function 'where' may not be changed.
qsort1.diff
qsort2.diff
query_decl.diff
resid1.diff
resid2.diff
retract1.diff
*** Garbage Collections: 1
retract2.diff
*** Garbage Collections: 1
retract3.diff
*** Garbage Collections: 1
retract4.diff
*** Garbage Collections: 1
retract5.diff
retract6.diff
retract7.diff
rootsort.diff
schedule1.diff
*** Garbage Collections: 7
schedule2.diff
school.diff
self.diff
share.diff
simp_eqn.diff
*** Garbage Collections: 7
smalltwente.diff
smm.diff
soap.diff
sortcycle1.diff
*** Error: there is a cycle in the sort hierarchy.
*** Cycle: [j <| n <| a <| b <| d <| f <| j]
sortcycle2.diff
*** Error: there is a cycle in the sort hierarchy.
*** Cycle: [j <| n <| a <| b <| d <| f <| j]
sortcycle3.diff
sortcycle4.diff
*** Error: there is a cycle in the sort hierarchy.
*** Cycle: [j <| o <| e <| j]
sortcycle5.diff
*** Garbage Collections: 2
str.diff
str2psi.diff
strbug.diff
*** Garbage Collections: 6
streams.diff
subs.diff
*** Garbage Collections: 12
subsort.diff
t3201.diff
t3202.diff
t3203.diff
t3204.diff
t3205.diff
t3206.diff
t3207.diff
*** Error: division by zero in 3 / 0.
t4105.diff
t4202.diff
t4502.diff
t4505.diff
*** Garbage Collections: 6

*** OUT OF MEMORY ***

t8101.diff
*** Error: square root of negative number in sqrt(-1).
*** Error: division by zero in 1 / 0.
*** Error: logarithm of zero in log(0).
t8102.diff
*** Warning: extending definition of sort 'real'.
t8103.diff
t8104.diff
*** Warning: extending definition of sort 'int'.
*** Garbage Collections: 3

*** OUT OF MEMORY ***

t8105.diff
tan.diff
test00.diff
test03.diff
test04.diff
*** Garbage Collections: 2
titi.diff
tri_ins.diff
tripeq.diff
unify.diff
varlevel.diff
varlevel2.diff
wam.diff
*** Garbage Collections: 4
wambruno.diff
wlife.diff
write.diff
writeq.diff
*** Garbage Collections: 2
z_comb.diff
*** Garbage Collections: 70
z_cut2.diff
*** Garbage Collections: 10
z_gcsimp.diff
*** Garbage Collections: 7
*** Error: psi-term too large -- get a bigger Life!
z_gctoken1.diff
*** Garbage Collections: 6
z_gctoken2.diff
*** Garbage Collections: 2
z_maurice.diff
*** Garbage Collections: 23

*** OUT OF MEMORY ***

z_nestdepth.diff
*** Garbage Collections: 4
z_power_4.diff
*** Garbage Collections: 234
z_pyth_gc.diff
