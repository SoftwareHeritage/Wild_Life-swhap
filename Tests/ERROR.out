addtab.diff
anyorder.diff
apply.diff
arith1.diff
arith2.diff
arith3.diff
*** Error: division by zero in B:0 / B.
*** Error: division by zero in real / 0.
*** Error: division by zero in 2 / 0.
arith4.diff
arith5.diff
arith6.diff
*** Error: division by zero in real / 0.
*** Error: division by zero in 0 / 0.
*** Error: division by zero in 1 / 0.
arith7.diff
*** Error: division by zero in real / 0.
*** Error: division by zero in real / 0.
arith8.diff
arith9.diff
assert1.diff
assert2.diff
assert3.diff
assert4.diff
bagof1.diff
bagof2.diff
bagof3.diff
bagof4.diff
bagof5.diff
bigstr.diff
bk_assign.diff
boites1.diff
boites2.diff
bruno_disj.diff
cb.diff
chapla.diff
choice.diff
clause.diff
colon.diff
compose.diff
constr.diff
copy1.diff
copy2.diff
copy3.diff
curry1.diff
curry2.diff
curry3.diff
curry4.diff
curry5.diff
cut1.diff
cut2.diff
cut3.diff
cut4.diff
cycles1.diff
cycles2.diff
decl.diff
delay.diff
dichotomy.diff
dict.diff
disequality1.diff
disequality2.diff
disequality3.diff
disequality4.diff
disequality5.diff
disequality6.diff
disequality7.diff
disequality8.diff
disequality9.diff
disj1.diff
disj2.diff
disj3.diff
disj4.diff
disj5.diff
disj6.diff
dissolve.diff
*** Error: the user-defined predicate 'p' may not be changed.
*** Error: the user-defined predicate 'p' may not be changed.
*** Error: the user-defined predicate 'p' may not be changed.
*** Error: the user-defined predicate 'p' may not be changed.
dynstat.diff
*** Syntax error near line 4: bad expression.
eoftest.diff
fact.diff
fail.diff
feature.diff
freeze1.diff
*** Error: the argument @ < @ of freeze must be a predicate.
*** Error: the argument @ = @ of freeze must be user-defined.
freeze2.diff
freeze3.diff
*** Error: the argument var(@~) of freeze must be user-defined.
*** Error: the argument @~ < @ of freeze must be a predicate.
*** Error: the argument write(@) of freeze must be user-defined.
freeze4.diff
function1.diff
function2.diff
fusil.diff
gauss.diff
gctest.diff
genint.diff
glb.diff
glblub.diff
grammars.diff
hamming.diff
hanoi.diff
inherit.diff
intabs.diff
*** Warning: extending definition of sort 'int'.
*** Warning: extending definition of sort 'int'.
inttrace.diff
isatest.diff
junk.diff
lefun1.diff
lefun2.diff
list_bda.diff
listing1.diff
*** Warning: extending definition of sort 'int'.
*** Warning: extending definition of sort 'zero'.
*** Warning: extending definition of sort 'int'.
listing2.diff
loadquery.diff
long.diff
longstr.diff
loop.diff
lub.diff
machine.diff
magic.diff
makestr.diff
mort.diff
*** Error: [@|@] should be a nil-terminated list in mresiduate.
*** Error: {@;@;@} should be a nil-terminated list in mresiduate.
mresid.diff
nesting.diff
nicetest.diff
nonstrict1.diff
nonstrict2.diff
nonstrict3.diff
nonstrict4.diff
nonstrict5.diff
numeric.diff
op1.diff
*** Error: numbers or strings may not be operators in '*op*'(500,xfy,"aaa").
*** Error: numbers or strings may not be operators in '*op*'(500,xfy,123).
*** Error: numbers or strings may not be operators in '*op*'(500,xfy,123.3).
*** Error: bad operator kind 'xff'.
*** Error: precedence must range from 1 to 1200 in '*op*'(5000,xfy,asd).
*** Error: precedence must range from 1 to 1200 in '*op*'(0,xfy,asd).
*** Error: precedence must be a positive integer in '*op*'(- 234,xfy,asd).
*** Error: precedence must be a positive integer in '*op*'(abc,xfy,asd).
*** Error: argument missing in '*op*'(500,xfy).
*** Error: argument missing in '*op*'(500,3 => asd).
*** Error: argument missing in '*op*'(2 => xfy,3 => asd).
op2.diff
order.diff
parents.diff
parse.diff
part.diff
person1.diff
person2.diff
*** Warning: extending definition of sort 'int'.
posint.diff
preparser.diff
*** Warning: extending definition of sort 'int'.
prime.diff
print1.diff
print2.diff
*** Error: argument in print_depth(...) must be positive or zero.
*** Error: argument in print_depth(-1) must be positive or zero.
print_depth1.diff
print_depth2.diff
printvar.diff
prove.diff
pyth.diff
*** Error: the user-defined function 'where' may not be changed.
qsort1.diff
qsort2.diff
query_decl.diff
resid1.diff
resid2.diff
retract1.diff
retract2.diff
retract3.diff
retract4.diff
retract5.diff
retract6.diff
retract7.diff
rootsort.diff
schedule1.diff
schedule2.diff
school.diff
self.diff
share.diff
simp_eqn.diff
smalltwente.diff
smm.diff
soap.diff
sortcycle1.diff
*** Error: there is a cycle in the sort hierarchy.
*** Cycle: [j <| n <| a <| b <| d <| f <| j]
sortcycle2.diff
*** Error: there is a cycle in the sort hierarchy.
*** Cycle: [j <| n <| a <| b <| d <| f <| j]
sortcycle3.diff
sortcycle4.diff
*** Error: there is a cycle in the sort hierarchy.
*** Cycle: [j <| o <| e <| j]
sortcycle5.diff
str.diff
str2psi.diff
strbug.diff
streams.diff
subs.diff
subsort.diff
t3201.diff
t3202.diff
t3203.diff
t3204.diff
t3205.diff
t3206.diff
t3207.diff
*** Error: division by zero in 3 / 0.
t4105.diff
t4202.diff
t4502.diff
t4505.diff

*** OUT OF MEMORY ***

*** Abort
t8101.diff
*** Error: square root of negative number in sqrt(-1).
*** Error: division by zero in 1 / 0.
*** Error: logarithm of zero in log(0).
t8102.diff
*** Warning: extending definition of sort 'real'.
t8103.diff
t8104.diff
*** Warning: extending definition of sort 'int'.

*** OUT OF MEMORY ***

*** Abort
t8105.diff
tan.diff
test00.diff
test03.diff
test04.diff
titi.diff
tri_ins.diff
tripeq.diff
unify.diff
varlevel.diff
varlevel2.diff
wam.diff
wambruno.diff
wlife.diff
write.diff
writeq.diff
z_comb.diff
z_cut2.diff
z_gcsimp.diff
*** Error: psi-term too large -- get a bigger Life!
*** Abort
z_gctoken1.diff
z_gctoken2.diff
z_maurice.diff

*** OUT OF MEMORY ***

*** Abort
z_nestdepth.diff
z_power_4.diff
z_pyth_gc.diff
